lib/twirp/encoder.ex

```diff
@@ -77,4 +74,15 @@ defmodule Twirp.Encoder do
   end
 
   defp strip_structs(any), do: any
+
+  defp to_atom_keys(map) when is_map(map) do
+    map
+    |> Enum.map(fn {key, value} ->
+      k = if is_binary(key), do: String.to_existing_atom(key), else: key
+      v = to_atom_keys(value)
+      {k, v}
+    end)
+    |> Enum.into(%{})
```

Comment:
you can avoid extra pass on the enum by either a comprehension:

```suggestion
    for {key, value} <- map, into: %{} do
      k = if is_binary(key), do: String.to_existing_atom(key), else: key
      v = to_atom_keys(value)
      {k, v}
    end
```

or using `Map.new/2` :)

Reply:
Oh fancy. I always forget I can do that with `for`.